handler/ratelimit/ring_buffer_internal_test.go

```diff
@@ -0,0 +1,77 @@
+package ratelimit
+
+import (
+	"testing"
+	"time"
+)
+
+func TestRingBuffer_New(t *testing.T) {
+	r := newRingBuffer(3)
+
+	if len(r.buf) != 3 {
+		t.Errorf("Unexpected r.buf: %#v", r.buf)
+	}
+	if r.len != 3 {
+		t.Errorf("Unexpected r.len: %d", r.len)
+	}
+	if r.r != 0 {
+		t.Errorf("Unexpected r.r: %d", r.r)
+	}
+	if r.w != 2 {
+		t.Errorf("Unexpected r.w: %d", r.w)
+	}
+}
+
+func TestRingBuffer_Put(t *testing.T) {
```

Comment:
* `put()` different times, so that proper positions of the entries can be testet
* have at least one complete rotation, so that at least one previously `put` time is eventually dropped
* also test `get()`

Reply:
Done